operator-common/src/main/java/io/strimzi/operator/cluster/model/Ca.java

```diff
@@ -794,32 +800,66 @@ public int keyGeneration() {
         return INIT_GENERATION;
     }
 
-    private int removeExpiredCerts(Map<String, String> newData) {
+    /**
+     * Remove old certificates that are stored in the CA Secret matching the "ca-YYYY-MM-DDTHH-MM-SSZ.crt" naming pattern.
+     * NOTE: mostly used when a CA certificate is renewed by replacing the key
+     */
+    public void maybeDeleteOldCerts() {
+        this.caCertsRemoved = removeCerts(this.caCertSecret.getData(), entry -> OLD_CA_CERT_PATTERN.matcher(entry.getKey()).matches()) > 0;
+        if (this.caCertsRemoved) {
+            LOGGER.infoCr(reconciliation, "{}: Old CA certificates removed", this);
+            /*
+            this.caCertSecret = new SecretBuilder(this.caCertSecret)
+                    .withData(this.caCertSecret.getData())
+                    .build();
+             */
```

Comment:
Is this some left over which can be deleted?

Reply:
Yeah :-(